File: src/main/java/wzq/jcstress/plugin/configuration/JCStressSettingsEditor.java
package wzq.jcstress.plugin.configuration;

import javax.swing.JComponent;

import com.intellij.execution.ui.CommonJavaParametersPanel;
import com.intellij.openapi.options.SettingsEditor;
import org.jetbrains.annotations.NotNull;

/**
 * @author 吴志强
 * @date 2022/8/24
 */
public class JCStressSettingsEditor extends SettingsEditor<JCStressRunConfiguration> {

    private final CommonJavaParametersPanel editor = new CommonJavaParametersPanel();

    @Override
    protected void resetEditorFrom(@NotNull JCStressRunConfiguration runConfiguration) {
        this.editor.reset(runConfiguration);
    }

    @Override
    protected void applyEditorTo(@NotNull JCStressRunConfiguration runConfiguration) {
        this.editor.applyTo(runConfiguration);
    }

    @Override
    protected @NotNull JComponent createEditor() {
        return this.editor;
    }
}

File: src/main/kotlin/wzq/jcstress/plugin/JCStressSettingsEditor.kt
package wzq.jcstress.plugin

import com.intellij.execution.ui.CommonJavaParametersPanel
import com.intellij.openapi.options.SettingsEditor
import javax.swing.JComponent

/**
 * @author 吴志强
 */
class JCStressSettingsEditor : SettingsEditor<JCStressRunConfiguration>() {
    private val editor = CommonJavaParametersPanel()

    override fun resetEditorFrom(configuration: JCStressRunConfiguration) {
        this.editor.reset(configuration)
    }

    override fun applyEditorTo(configuration: JCStressRunConfiguration) {
        this.applyTo(configuration)
    }

    override fun createEditor(): JComponent = this.editor
}
